metrics/perf_monitor/src/counters.rs

```diff
@@ -79,3 +80,49 @@ pub struct CountersSnapshot {
     pub disk_io_read_per_sec: f64,
     pub disk_io_write_per_sec: f64,
 }
+
+fn to_human_readable(mut number_to_format: f64, precision: usize, suffix: &str) -> String {
+    let units = vec!["", "K", "M", "G", "T", "P", "E"];
+    let mut found_unit = "";
+
+    for unit in units {
+        if number_to_format < 1000.0 {
+            found_unit = unit;
+            break;
+        } else {
+            number_to_format /= 1000.0
+        }
+    }
+
+    format!("{number_to_format:.precision$}{}{}", found_unit, suffix)
+}
+
+impl CountersSnapshot {
+    pub fn log_metrics(&self, service_name: &str) {
```

Comment:
Implement it as Display trait implementation

Reply:
Does implementing Display work with splitting the log in two lines?

Reply:
Sure,
`writeln` first line, `write` second 
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=afc67a6e6fc4a953db4ba458fdc7f7b1

Reply:
This is how it'd look like with the `impl Display`:

```
2023-12-23 15:44:43.605-07:00 [TRACE] Simulation time: 115.266  Generated 956 blocks
2023-12-23 15:44:45.136-07:00 [TRACE] Performance Metrics
Process Metrics: RSS: 504184832 (504.18MB), VIRT: 420408082432 (420.41GB), cores: 16, cpu usage (per core): 0.7272727272727273
Disk IO Metrics: FD: 25, read: 172032 (172KB), write: 949473280 (949MB), read rate: 3685.662867426515 (4KB/s), write rate: 57424675.064987004 (57MB/s)
2023-12-23 15:44:45.713-07:00 [TRACE] Simulation time: 119.167  Generated 1000 blocks
```

I added "Performance Metrics" as the first line so that both process and disk IO metrics appear together in new lines.